specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-11-01-preview/TuningOptions.json

```diff
@@ -0,0 +1,456 @@
+      "get": {
+        "tags": [
+          "TuningOptions"
+        ],
+        "operationId": "TuningOptions_ListByServer",
+        "x-ms-examples": {
+          "TuningOptions_ListByServer": {
+            "$ref": "./examples/Tuning_ListTuningOptions.json"
+          }
+        },
+        "description": "Retrieve the list of available tuning options.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../types/common-types.json#/parameters/ServerNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/TuningOptionsListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ImpactRecord": {
+      "type": "object",
+      "properties": {
+        "dimensionName": {
+          "type": "string",
+          "description": "Dimension name"
+        },
+        "unit": {
+          "type": "string",
+          "description": "Dimension unit"
+        },
+        "queryId": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Optional property that can be used to store the QueryId if the metric is for a specific query."
+        },
+        "absoluteValue": {
+          "type": "number",
+          "format": "double",
+          "description": "Absolute value"
+        }
+      },
+      "description": "Stores property that features impact on some metric if this recommended action is applied."
+    },
+    "IndexRecommendationDetails": {
+      "type": "object",
+      "properties": {
+        "databaseName": {
+          "type": "string",
+          "description": "Database name."
+        },
+        "schema": {
+          "type": "string",
+          "description": "Schema name."
+        },
+        "table": {
+          "type": "string",
+          "description": "Table name."
+        },
+        "indexType": {
+          "type": "string",
+          "description": "Index type."
+        },
+        "indexName": {
+          "type": "string",
+          "description": "Index name."
+        },
+        "indexColumns": {
```

Comment:
>indexColumns [](http://example.com/codeflow?start=9&length=12)

Collections should be modeled as arrays.

Reply:
This is not a collection; the response returns a string:
`"indexColumns": "\"lineitem\".\"l_partkey\";\"lineitem\".\"l_quantity\""`

Reply:
The feedback is that is should be implemented as an array of strings, rather than a single string that requires clients to parse it on semicolon separators. The property name and description clearly implies it's a collection of index columns.

Reply:
when the rest API was reviewed (see ref here https://github.com/Azure/azure-rest-api-specs/pull/29295/files/6771c9dc5d2260a26a422c047c447e29e7f6818c#r1634739420) it was accepted to add in the description that the values are semi-colon separated. This request comes at a very late stage. Also there are other RPs following the same practice, for instace SQL DB, ref here: https://learn.microsoft.com/en-us/rest/api/sql/database-recommended-actions/list-by-database-advisor?view=rest-sql-2021-11-01&tabs=HTTP#list-of-database-recommended-actions  . SQL DB already has swagger, sdk, powershel GAed with this behavior

Reply:
Discussed this with @nasc17 and the RP team in office hours and agreed to sign off this PR according to the following:
1. RP team will remove the "includedColumns" property so the change isn't needed.
2. RP team will implement the "indexColumns" property as an array in a future API version.